### PR TITLE
Pi+ finish coxter proof

### LINKED ISSUE #12: Finish the proof of equivalence between Coxeter and Lehmer

### PR DESCRIPTION
**Pull request by inexxt**
TODO:
 - [x] Finish Lehmer proof
 - [x] Finish CanonicalForm proof
 - [x] Finish Arithmetic

until no holes or postulates remain.

Fixes #12

### CODE REVIEW

File: Pi+/Coxeter/Lehmer.agda [hunk 1]
```diff
@@ -87,40 +87,234 @@ lemma-l++2++r a b (x :: l1) r1 (x₁ :: l2) r2 p with lemma-l++2++r a b l1 r1 l2
...
+            S (S (r₂ + S n ∸ r₂))
+          ≡⟨ ap (λ e -> S (S e)) (cut-tail snd₁) ⟩
+            S (S k₁)
+          ≤⟨ ps ⟩
+            n₁
+          ≤∎
+  in  >>-⊥ (S n) n₁ (≤-down (≤-down lemma)) (immersion (CanS cl {r₁} x₁)) l nil (immersion->> (CanS cl {r₁} x₁)) fst₁
+final≅-Lehmer {S (S n)} (CanS (CanS cl {r₁} x₁) {r₂} x₂) m mf defm (long≅ {n₁} k l r .m .mf defm₁ defmf) = 
+  let lemma = transport (λ e -> l ++ S (k + n₁) :: k + n₁ :: (n₁ ↓ k ++ S (k + n₁) :: e) == l ++ S (k + n₁) :: k + n₁ :: (n₁ ↓ k ++ S (k + n₁) :: rev (rev r))) (! rev-rev) idp
+  in  immersion-long-lemma (CanS (CanS cl x₁) x₂) k n₁ l (rev r) ( (! defm) ∙ defm₁ ∙ lemma)
```
Comment (inexxt) [inline Pi+/Coxeter/Lehmer.agda:311]: I don't know why Agda doesn't accept the termination condition - I am doing structural recursion on the first parameter.
@vikraman @JacquesCarette @sabry Any clue?

Comment (vikraman) [reply Pi+/Coxeter/Lehmer.agda:311]: Setting `--termination-depth=2` seems to work!

Comment (inexxt) [reply Pi+/Coxeter/Lehmer.agda:311]: Thanks! Fixed.

File: Pi+/Coxeter/Lists.agda [hunk 2]
```diff
@@ -85,6 +85,21 @@ head+tail idp idp = idp
...
+cut-last {a1} {a2} {x :: x₁ :: l1} {nil} ()
+cut-last {a1} {a2} {x :: l1} {x₁ :: l2} p = head+tail (cut-tail p) (cut-last (cut-head p))
+
+cut-prefix : {a1 a2 : ℕ} -> {l1 l2 : List} -> (l1 ++ [ a1 ] == l2 ++ [ a2 ]) -> (a1 == a2)
+cut-prefix {a1} {.a1} {nil} {nil} idp = idp
+cut-prefix {a1} {a2} {nil} {x :: nil} ()
+cut-prefix {a1} {a2} {nil} {x :: x₁ :: l2} ()
+cut-prefix {a1} {a2} {x :: nil} {nil} ()
+cut-prefix {a1} {a2} {x :: x₁ :: l1} {nil} ()
+cut-prefix {a1} {a2} {x :: l1} {x₁ :: l2} p = cut-prefix (cut-head p)
```
Comment (vikraman) [inline Pi+/Coxeter/Lists.agda:103]: These lemmas have better proofs using the path space of lists, but we can ignore that for now.

Comment (inexxt) [reply Pi+/Coxeter/Lists.agda:103]: You can change that, I will be happy to learn!

**Pull request merged**